assembly-multiuser/assembly-wsmaster-war/src/main/webapp/keycloak/OIDCKeycloak.js

```diff
@@ -0,0 +1,1371 @@
+                }
+
+                return { baseUri: baseUri, queryString: queryString, fragmentString: fragmentString };
+            }
+
+            var parseParams = function(paramString) {
+                var result = {};
+                var params = paramString.split('&');
+                for (var i = 0; i < params.length; i++) {
+                    var p = params[i].split('=');
+                    var paramName = decodeURIComponent(p[0]);
+                    var paramValue = decodeURIComponent(p[1]);
+                    result[paramName] = paramValue;
+                }
+                return result;
+            }
+
+            var handleQueryParam = function(paramName, paramValue, oauth) {
+                var supportedOAuthParams = [ 'code', 'state', 'error', 'error_description' ];
+
+                for (var i = 0 ; i< supportedOAuthParams.length ; i++) {
+                    if (paramName === supportedOAuthParams[i]) {
+                        oauth[paramName] = paramValue;
+                        return true;
+                    }
+                }
+                return false;
+            }
+
+
+            parser.parseUri = function() {
+                var parsedUri = initialParse();
+
+                var queryParams = {};
+                if (parsedUri.queryString) {
+                    queryParams = parseParams(parsedUri.queryString);
+                }
+
+                var oauth = { newUrl: parsedUri.baseUri };
+                for (var param in queryParams) {
+                    switch (param) {
+                        case 'redirect_fragment':
+                            oauth.fragment = queryParams[param];
+                            break;
+                        default:
+                            if (responseMode != 'query' || !handleQueryParam(param, queryParams[param], oauth)) {
+                                oauth.newUrl += (oauth.newUrl.indexOf('?') == -1 ? '?' : '&') + param + '=' + encodeURIComponent(queryParams[param]);
+                            }
+                            break;
+                    }
+                }
+
+                if (responseMode === 'fragment') {
+                    var fragmentParams = {};
+                    if (parsedUri.fragmentString) {
+                        fragmentParams = parseParams(parsedUri.fragmentString);
+                    }
+                    for (var param in fragmentParams) {
+                        oauth[param] = fragmentParams[param];
+                    }
+                }
+
+                return oauth;
+            }
+        }
+
+    }
+
+    if ( typeof module === "object" && module && typeof module.exports === "object" ) {
+        module.exports = Keycloak;
+    } else {
+        window.Keycloak = Keycloak;
+
+        if ( typeof define === "function" && define.amd ) {
+            define( "keycloak", [], function () { return Keycloak; } );
+        }
+    }
+})( window );
```

Comment:
how is unit tested that JS ?

Reply:
That JS is not unit-tested for now TBH since it's mainly a modification of the `keycloak.js` Javascript adapter file. The next step (not a critical priority though) is to open a PR to the Keycloak repo to propose these changes, and have them tested in the context of the Keycloak project. I should be able to open the PR this week, and should be able to discuss with Keycloak team member about this at the end of the week.

Reply:
@davidfestal maybe add a comment on the top of the file that it's a change from upstream keycloak and/or if possible to have a diff to maintain it over new versions of keycloak ?

Reply:
well, I think that for now I can add a comment at the top of the file. Not sure about the diff, which would make it hard to maintain IMO.

The idea in fact is that, once it is merged on the Keycloak side, Che could simply retrieve the keycloak.js through NPM since the Keycloak JS adapter is available as a standalone NPM package: https://www.npmjs.com/package/keycloak-js.

Then we won't have to manage a diff at all.

Reply:
@benoitf would this comment be sufficient for you : https://github.com/eclipse/che/pull/8614/commits/8237cd3b4115eb2355f0ae9ee707c2b1072a87e8 ?